deploy/lib/xquery/setup.xqy

```diff
@@ -4282,18 +4282,33 @@ declare function setup:get-configuration(
   $databases as xs:string*,
   $forests as xs:string*,
   $app-servers as xs:string*,
-  $user-ids as xs:unsignedLong*,
-  $role-ids as xs:unsignedLong*,
+  $users as xs:anySimpleType*,
+  $roles as xs:anySimpleType*,
   $mimetypes as xs:string*) as element()
 {
-  <configuration>
-    {setup:get-app-servers($app-servers)}
-    {setup:get-forests($forests)}
-    {setup:get-databases($databases)}
-    {setup:get-users($user-ids)}
-    {setup:get-roles($role-ids)}
-    {setup:get-mimetypes($mimetypes)}
-  </configuration>
+  let $user-configuration :=
+      typeswitch($users[1])
```

Comment:
The typeswitch looks nice, but doesn't work like this. I think you should use an old fashion if with a castable as test, and then actually typecast the args for setup:get-users. I would also recommend checking and casting to xs:unsignedLong, as that is what the get-users function expects..